msteams-platform/apps-in-teams-meetings/teams-live-share-how-to/how-to-custom-azure-fluid-relay.md

```diff
@@ -0,0 +1,142 @@
+  initialObjects: {
+    presence: EphemeralPresence,
+    ticTacToePositions: SharedMap,
+  },
+};
+const { container } = await client.joinContainer(schema);
+
+// ... ready to start app sync logic
+```
+
+# [TypeScript](#tab/typescript)
+
+```TypeScript
+import { TeamsFluidClient, EphemeralPresence, ITeamsFluidClientOptions } from "@microsoft/live-share";
+import { SharedMap } from "fluid-framework";
+import { AzureFunctionTokenProvider } from "@fluidframework/azure-client";
+
+// Define a custom connection for your app
+const clientProps: ITeamsFluidClientOptions = {
+  connection: {
+    tenantId: "MY_TENANT_ID",
+    tokenProvider: new AzureFunctionTokenProvider(
+      "MY_FUNCTION_ENDPOINT_URL" + "/api/GetAzureToken",
+      { userId: "userId", userName: "Test User" }
+    ),
+    endpoint: "MY_SERVICE_ENDPOINT_URL",
+    type: "remote",
+  },
+};
+// Join the Fluid container
+const client = new TeamsFluidClient(clientProps);
+const schema = {
+  initialObjects: {
+    presence: EphemeralPresence,
+    ticTacToePositions: SharedMap,
+  },
+};
+const { container } = await client.joinContainer(schema);
+
+// ... ready to start app sync logic
+```
+
+---
+
+## Why use a custom Azure Fluid Relay service?
+
+While most developers will find our free service meets all their requirements, there may be cases where using your own Azure Fluid Relay service for your Live Share app is ideal.
+
+Consider using a custom AFR service connection if you:
+
+- Require storage of data in Fluid containers beyond the lifetime of a meeting.
+- Transmit sensitive data through the service that requires a custom security policy.
+- Develop features through Fluid Framework (e.g., `SharedMap`) for your application outside of Teams.
+
+## Why use Live Share with your custom service?
+
+Azure Fluid Relay is designed to work with any web-based application, meaning it works with or without Microsoft Teams. That raises an important question: if I build my own Azure Fluid Relay service, do I still need Live Share?
+
+Live Share has features that are beneficial to common meeting scenarios that augment other features in your app, including:
+
+- Container mapping
+- Ephemeral objects & role verification
+- Media synchronization
+
+### Container mapping
+
+Live Share's `TeamsFluidClient` class is responsible for mapping a unique meeting identifier to your Fluid containers, which ensures that all meeting participants join the same container. As part of this process, the client will attempt to connect to a `containerId` mapped to the meeting should one already exist. If one does not exist, the `AzureClient` is used to create a container using your `AzureConnectionConfig` and then relay the `containerId` to other meeting participants.
+
+If your app already has a mechanism for creating Fluid containers and sharing them to other members -- such as by inserting the `containerId` into the URL shared to the meeting stage -- then this may not be necessary for your app.
```

Comment:
```suggestion
Live Share's `TeamsFluidClient` class is responsible for mapping a unique meeting identifier to your Fluid containers, which ensures that all meeting participants join the same container. As part of this process, the client attempts to connect to a `containerId` mapped to the meeting that one already exists. If one doesn't exist, the `AzureClient` is used to create a container using your `AzureConnectionConfig` and then relay the `containerId` to other meeting participants.

If your app already has a mechanism for creating Fluid containers and sharing them to other members, such as by inserting the `containerId` into the URL shared to the meeting stage, then this may not be necessary for your app.
```
Please verify for technical accuracy.

Reply:
Updated as mentioned.

Reply:
Updated as mentioned.

Reply:
Updated as mentioned.